test/system/test_cvp_api.py

```diff
@@ -0,0 +1,395 @@
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# 'AS IS' AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL ARISTA NETWORKS
+# BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR
+# BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+# WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE
+# OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN
+# IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+
+''' System test for the CvpApi class
+
+    Requirements for CVP Node:
+    1) Test has dedicated access to the CVP node.
+    2) Contains at least one device in a container.
+    3) Container or device has at least one configlet applied.
+
+    After running tests the device may show that it is out of compliance.
+    This is due to applying configlets and then canceling the tasks.
+'''
+import os
+import re
+import sys
+import time
+import unittest
+
+from cvprac.cvp_client import CvpClient
+from cvprac.cvp_client_errors import CvpApiError
+
+sys.path.append(os.path.join(os.path.dirname(__file__), '../lib'))
+from systestlib import DutSystemTest
+
+class TestCvpClient(DutSystemTest):
+    ''' Test cases for the CvpClient class.
+    '''
+    # pylint: disable=too-many-public-methods
+    def setUp(self):
+        ''' Instantiate the CvpClient class and connect to the CVP node.
+            Log messages to the /tmp/TestCvpClient.log
+        '''
+        super(TestCvpClient, self).setUp()
+        self.clnt = CvpClient(filename='/tmp/TestCvpClient.log')
+        self.assertIsNotNone(self.clnt)
+        dut = self.duts[0]
+        self.clnt.connect([dut['node']], dut['username'], dut['password'])
+        self.api = self.clnt.api
+        self.assertIsNotNone(self.api)
+
+        # Verify that there is at least one device in the inventory
+        err_msg = 'CVP node must contain at least one device'
+        result = self.api.get_inventory()
+        self.assertIsNotNone(result)
+        self.assertGreaterEqual(len(result), 1, msg=err_msg)
+        self.device = result[0]
+
+        # Get the container for the device on the list and
+        # use that container as the parent container.
+        result = self.api.search_topology(self.device['fqdn'])
+        self.assertIsNotNone(result)
+        dev_container = result['netElementContainerList']
+        self.assertGreaterEqual(len(dev_container), 1)
+        info = dev_container[0]
+        result = self.api.search_topology(info['containerName'])
+        self.assertIsNotNone(result)
+        self.container = result['containerList'][0]
+
+        # Get the configlets assigned to the device.  There must be at least 1.
```

Comment:
You really have to have at least one configlet that sets up AAA and management IP address. If I create one then I have to set that up otherwise applying the configlet would fail because it would be wiping out the user account and management IP on the switch. The tests give an accurate error for missing configlet or node but not for incorrect user credentials. I will add a check at setup for that.